### PR TITLE
Gh-2840: MapStore MatchedVertex inconsistent

### LINKED ISSUE #2840: Map store GetAllElements returns matchedVertex on Edge results
**Describe the bug**
After a GetAllElements operation when the result is an Edge, the map store returns matchedVertex but the accumulo store does not. Which is correct behaviour and why the difference?

**To Reproduce**
Add an Edge to a map store backed graph. Do GetAllElements. Result contains `"matchedVertex": "SOURCE"`

**Expected behavior**
It probably should not as this is not very useful, it will always match on source for GetAllElements

**Additional context**
Added a branch with temporary test to show behaviour: https://github.com/gchq/Gaffer/commit/6904676b54deaa8d8172b6f58a698954c5dcf5e5

### PR DESCRIPTION
**Pull request by j69772**
# Related issue

- Resolve #2840

### CODE REVIEW

File: core/operation/src/main/java/uk/gov/gchq/gaffer/operation/data/EdgeSeed.java [hunk 1]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2016-2023 Crown Copyright
+ * Copyright 2016-2024 Crown Copyright
```
Comment (p29876) [inline core/operation/src/main/java/uk/gov/gchq/gaffer/operation/data/EdgeSeed.java:2]: Has there been a change to this class this year?

File: store-implementation/map-store/src/main/java/uk/gov/gchq/gaffer/mapstore/impl/AddElementsHandler.java [hunk 2]
```diff
@@ -150,9 +150,10 @@ private void updateElementIndex(final Element element, final MapImpl mapImpl) {
             mapImpl.addIndex(entitySeed, element);
         } else {
             final Edge edge = (Edge) element;
-            edge.setIdentifiers(edge.getSource(), edge.getDestination(), edge.isDirected(), MatchedVertex.SOURCE);
+            final Edge sourceMatchedEdge = new Edge(edge.getGroup(), edge.getSource(), edge.getDestination(), edge.isDirected(), MatchedVertex.SOURCE, edge.getProperties());
+
             final EntitySeed sourceEntitySeed = new EntitySeed(edge.getSource());
-            mapImpl.addIndex(sourceEntitySeed, edge);
+            mapImpl.addIndex(sourceEntitySeed, sourceMatchedEdge);
```
Comment (p29876) [inline store-implementation/map-store/src/main/java/uk/gov/gchq/gaffer/mapstore/impl/AddElementsHandler.java:156]: :fire:

File: store-implementation/map-store/src/test/java/uk/gov/gchq/gaffer/mapstore/impl/GetAllElementsHandlerTest.java [hunk 3]
```diff
@@ -57,7 +60,16 @@ public class GetAllElementsHandlerTest {
     static final String PROPERTY1 = "property1";
     static final String PROPERTY2 = "property2";
     static final String COUNT = "count";
+    public static final Context CONTEXT = new Context(new User("user"));
     private static final int NUM_LOOPS = 10;
+    public static final String EXPECTED = "{\n" +
```
[the remark this answers is not in the document]
Comment (j69772) [reply store-implementation/map-store/src/test/java/uk/gov/gchq/gaffer/mapstore/impl/GetAllElementsHandlerTest.java:65]: Thank you - Will do.

**Comment by p29876** [comment]
Ready for re-review
Can the mergers please ensure that the squashed commit message contains a reference to #3314 for the history

**Review by cn337131 — DISMISSED** [review]

**Review by rj77259 — DISMISSED** [review]

**Review by cn337131 — APPROVED** [review]

**Pull request merged**